### ISSUE TITLE
markevery only accepts builtin integers, not numpy integers

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#8057]
mpl 2.0/py 3.6
`plot(range(10), ".", markevery=3)` works, but `plot(range(10), ".", markevery=np.int_(3))'` fails with
```
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/matplotlib/backends/backend_qt5agg.py", line 197, in __draw_idle_agg
    FigureCanvasAgg.draw(self)
  File "/usr/lib/python3.6/site-packages/matplotlib/backends/backend_agg.py", line 464, in draw
    self.figure.draw(self.renderer)
  File "/usr/lib/python3.6/site-packages/matplotlib/artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/matplotlib/figure.py", line 1143, in draw
    renderer, self, dsu, self.suppressComposite)
  File "/usr/lib/python3.6/site-packages/matplotlib/image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "/usr/lib/python3.6/site-packages/matplotlib/artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/matplotlib/axes/_base.py", line 2409, in draw
    mimage._draw_list_compositing_images(renderer, self, dsu)
  File "/usr/lib/python3.6/site-packages/matplotlib/image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "/usr/lib/python3.6/site-packages/matplotlib/artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/matplotlib/lines.py", line 851, in draw
    affine, self.axes.transAxes)
  File "/usr/lib/python3.6/site-packages/matplotlib/lines.py", line 233, in _mark_every_path
    'markevery=%s' % (markevery,))
ValueError: Value of `markevery` is not recognized; markevery=3
```

This is due to the use of `isinstance(..., int)` in the relevant codepath.

**Comment by tacaswell** [comment]
Fixing this looks sutble, there is a bunch of explicit type checking.

**Comment by phobson** [comment]
closed via #8485
